scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+
+            // Download single file
+            filesRelativePathName.add(downloadSingleFile(channelSftp, sftpRemoteRelativePath, localFilePath));
+
+            // if the file is a zip and sftpextractarchive == true then extract the archive
+            if (sftpExtractArchive && sftpRemoteRelativePath.endsWith(".zip")) {
+                getOut().println("Decompressing archive: " + localFilePath);
+                ZipUtil.unpack(new File(localFilePath), new File(sftpLocalRelativePath));
+                getOut().println("Archive decompressed successfully at: " + sftpLocalRelativePath);
+            }
+
+            return filesRelativePathName;
+        }
+
+        //Iterate through list of folder content
+        for (ChannelSftp.LsEntry item : fileAndFolderList) {
+
+            String remoteFilePath = Paths.get(sftpRemoteRelativePath, item.getFilename()).toString();
+            String localFilePath = Paths.get(sftpLocalRelativePath, item.getFilename()).toString();
+            File localFile = new File(localFilePath);
+
+            // Check if it is a file (not a directory).
+            if (!item.getAttrs().isDir()) {
+                filesRelativePathName.add(downloadSingleFile(channelSftp, remoteFilePath, localFilePath));
+
+            } else // skip '.' and '..'
+            if (!(".".equals(item.getFilename()) || "..".equals(item.getFilename()))) {
+                // Copy empty folder.
+                localFile.mkdirs();
+                // Enter found folder on server to read its contents and create locally.
+                filesRelativePathName.addAll(recursiveFolderDownload(channelSftp, remoteFilePath, localFilePath));
+            }
+        }
+        return filesRelativePathName;
+    }
+
+    /**
+     * This method checks if a directory exists in a remote SFTP server
+     *
+     * @param channelSftp
+     * @param remoteFilePath
+     * @return true if the remote directory exists, false otherwise
+     */
+    private boolean remoteDirExists(ChannelSftp channelSftp, String remoteFilePath) {
+        try {
+            return channelSftp.stat(remoteFilePath).isDir();
+        } catch (SftpException e) {
+            // An exception is raised if file/directory is not found.
+            return false;
+        }
+    }
+
+    /**
+     * This method checks if the given LS results correspond to a file or not.
+     *
+     * @param filesList
+     * @return
+     */
+    private boolean isRemoteFile(List<ChannelSftp.LsEntry> filesList) {
+        return (filesList.size() == 1 && !filesList.get(0).getAttrs().isDir());
+    }
+
+    /**
+     * This method recursively creates nonexisting nested folders
+     * Example /folder/sub1/sub2/sub3/sub...
+     *
+     * @param channelSftp
+     * @param path
+     * @throws SftpException
+     */
+    private void makeRemoteDirectories(ChannelSftp channelSftp, Path path) throws SftpException {
```

Comment:
Maybe change name to `createRemoteDirectories`?

Reply:
In fact, we prefer "make" than "create" because it is closer to mkdir (make dir).